File: Universal Windows Platform Solution/Phantom of the West/Phantom of the West/Visual Novel/Chapters/StoryChoice.cs
using Windows.ApplicationModel.Resources;

namespace Phantom_of_the_West.Visual_Novel.Chapters
{
	internal class StoryChoice : IStoryChoice
	{
		internal static StoryChoice FromResource(string stringName, int nextState)
		{
			string localizedString = GetLocalizedString(stringName);
			StoryChoice newChoice = new StoryChoice(localizedString, nextState);
			return newChoice;
		}

		internal static string GetLocalizedString(string stringName)
		{
			ResourceLoader resourceLoader = ResourceLoader.GetForViewIndependentUse("StoryText");
			string localizedString = resourceLoader.GetString(stringName);
			return localizedString;
		}

		public string Text
		{
			get;
			private set;
		}

		protected int NextState
		{
			get;
			set;
		}

		internal StoryChoice(string text, int nextState)
		{
			Text = text;
			NextState = nextState;
		}

		public virtual void Select()
		{
			PotWVN vn = PotWVN.MainVN;
			vn.GoToState(NextState);
		}
	}
}

File: Android Project/app/src/main/java/com/christopherraleigh/phantomofthewest/visual_novel/chapters/StoryChoice.java
package com.christopherraleigh.phantomofthewest.visual_novel.chapters;

import android.content.Context;

import com.christopherraleigh.phantomofthewest.visual_novel.IStoryChoice;
import com.christopherraleigh.phantomofthewest.visual_novel.PotWVN;

/**
 * Created by christopherraleigh on 2016-04-15.
 */
public class StoryChoice implements IStoryChoice {
    private int nextState;

    private int textID;

    public StoryChoice(int textID, int nextState) {
        this.textID = textID;
        this.nextState = nextState;
    }

    @Override
    public CharSequence getText(Context c) {
        return c.getText(textID);
    }

    @Override
    public void select() {
        PotWVN vn = PotWVN.getMainVN();
        vn.goToState(nextState);
    }
}
